components/nodemanager-service/pgdb/node-managers-process-node.go

```diff
@@ -103,6 +105,8 @@ func (db *DB) ProcessIncomingNode(node *manager.NodeMetadata) error {
 
 	lastContact := ptypes.TimestampString(node.LastContact)
 
+	// note: we shouldn't be able to ever get here without a node id, because this function is called from the
+	// ingestion process, and the ingestion process will reject a report with no node uuid
```

Comment:
Given that, might it make sense to warn or fail here instead?

Reply:
ah good point! yes!